pkg/zeek/value.go

```diff
@@ -43,6 +48,8 @@ type Value interface {
 	// the contained Values and true.  If this value is not a container,
 	// return an empty list and false.
 	Elements() ([]Value, bool)
+
+	Encode(zval.Encoding) zval.Encoding
```

Comment:
I realize the underlying primitive is append-to but I like the Encode/Decode metaphor better.  I could be persuaded if the is consensus one way or the other.

Definitely needs a comment.

Reply:
Having considered it's usage, I don't think Append makes sense because you are appending the thing inside the value, so you're not doing
```
zv = Append(zv, val.Zval())
```
but rather 
```
zv =  val.Encode(zv)
```
so append looks odd:
```
zv =  val.Append(zv)
```